File: app/src/main/java/com/sarcobjects/a500mts/android/ContextModule.java
package com.sarcobjects.a500mts.android;

import android.content.Context;

import com.sarcobjects.a500mts.BingSnapToRoad;
import com.sarcobjects.a500mts.SnapToRoad;

import dagger.Module;
import dagger.Provides;

@Module
public class ContextModule {

    Context context;

    public ContextModule(Context context) {
        this.context = context;
    }

    @Provides
    Context getContext() {
        return context;
    }

    @Provides
    SnapToRoad getSnapToRoad() {
        return new BingSnapToRoad(getContext());
    }
}

File: app/src/main/java/com/sarcobjects/a500mts/android/ContextModule.kt
package com.sarcobjects.a500mts.android

import android.content.Context
import com.sarcobjects.a500mts.BingSnapToRoad
import com.sarcobjects.a500mts.SnapToRoad
import dagger.Module
import dagger.Provides

@Module
class ContextModule(@get:Provides var context: Context) {

    @get:Provides
    val snapToRoad: SnapToRoad
        get() = BingSnapToRoad(context)

}